python/lsst/ts/ofc/controllers/base_controller.py

```diff
@@ -173,44 +199,64 @@ def aggregated_state(self):
         """
         return self.dof_state[self.ofc_data.dof_idx]
 
+    def reset_history(self) -> None:
+        """Reset the history of the controller."""
+        self.dof_state0 = self.dof_state
+        self.integral = (
+            self.setpoint[self.ofc_data.dof_idx] - self.dof_state[self.ofc_data.dof_idx]
+        )
+        self.previous_error = (
+            self.setpoint[self.ofc_data.dof_idx] - self.dof_state[self.ofc_data.dof_idx]
+        )
+
     def control_step(
         self,
         filter_name: str,
         dof_state: np.ndarray,
         sensor_names: list[str] | None = None,
     ) -> np.ndarray:
-        """Estimate uk in the basis of degree of freedom (DOF) with gain
-        compensation.
+        """Estimate uk in the basis of degree of freedom (DOF).
 
         Parameters
         ----------
         filter_name : `string`
             Name of the filter.
         dof_state : `numpy.ndarray`
             Optical state in the basis of DOF.
-        sensor_names : `list` of `string`
+        sensor_names : `list [string]`
             List of sensor names.
 
+        Raises
+        ------
+        NotImplementedError
+            Child class should implemented this.
+        """
+        raise NotImplementedError("Child class should implemented this.")
+
+    def pid_step(self, state: np.ndarray) -> np.ndarray:
+        """
```

Comment:
Please add the description of this function and consider to rename the function name to be `calculate_pid_step()` or `calc_pid_step()`.